SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -684,21 +742,64 @@ bool RunSchemaCompiler()
 	}
 }
 
-bool SpatialGDKGenerateSchema()
+bool SpatialGDKGenerateSchema(bool bSaveSchemaDatabase, bool bRunSchemaCompiler)
 {
-	ResetUsedNames();
+	// Generate Schema for classes loaded in memory.
+
+	if (!SpatialGDKGenerateSchemaForClasses(GetAllSupportedClasses()))
+	{
+		return false;
+	}
+
+	if (bSaveSchemaDatabase)
+	{
+		if (!SaveSchemaDatabase())
+		{
+			return false;
+		}
+	}
+
+	if (bRunSchemaCompiler)
+	{
+		return RunSchemaCompiler();
+	}
 
-	// Gets the classes currently loaded into memory.
-	SchemaGeneratedClasses = GetAllSupportedClasses();
-	SchemaGeneratedClasses.Sort();
+	return true;
+}
 
+bool SpatialGDKGenerateSchemaForClasses(const TSet<UClass*>& Classes)
+{
+	ResetUsedNames();
+	TSet<UClass*> NewClasses = FilterClasses(Classes);
```

Comment:
I believe you can just sort the set if you provide a predicate

Reply:
Done